pkgs/build-support/trivial-builders.nix

```diff
@@ -318,23 +321,46 @@ rec {
   symlinkJoin =
     args_@{ name
          , paths
+         , outputs ? [ "out" ]
          , preferLocalBuild ? true
          , allowSubstitutes ? false
          , postBuild ? ""
          , ...
          }:
     let
+      secondaryOutputs = builtins.filter (o: o != "out") outputs;
+      outputToPathsAttr = output: {
+        name = output + "Paths";
+        value = lib.flatten (map (p: p.${output} or []) paths);
+      };
       args = removeAttrs args_ [ "name" "postBuild" ]
         // {
           inherit preferLocalBuild allowSubstitutes;
-          passAsFile = [ "paths" ];
-        }; # pass the defaults
+          outputs = [ "out" ] ++ secondaryOutputs;
```

Comment:
This makes sure that `"out"` is always present in the list - the user could supply a list without it, but we always need it. I suppose we could just always add it and run `lib.unique` on the list instead.

Reply:
I see! does it ever make sense to have an `outputs` provided to `symlinkJoin` that does not contain `"out"`?

Maybe it would make more sense to have `secondaryOutputs` be an argument to symlinkJoin, and you have a guarantee that you can't ever not include `out` in the outputs

Reply:
No, it wouldn't make sense, since we're using `out` as our main output. Since `output` is an argument to `mkDerivation` and the expected name of the parameter, I'd prefer to keep it and only use `secondaryOutputs` internally.

Reply:
:+1: Sounds good. I can foresee some confusion by allowing an `output` without `"out"` (does `mkDerivation` allow the same?), what would the user even expect to happen? 

But it's such an edge case & doesn't make sense anyways, I think is good enough, if not a bit unintuitive.

Thank you for explaining!

Reply:
Yes, you don't need an `"out"` output, `mkDerivation` uses the first one in the list as the primary one. Some packages call their main output `"bin"` instead, for example. However, we need it, since we're using it to create a combined output of all main outputs (which can vary in name). So, unless we want to change that behavior, we should make sure we always get an `"out"` output.